### PR TITLE
Add support for "Help" questionnaire item control

### LINKED ISSUE #1528: Add support for "Help" questionnaire item control
**Is your feature request related to a problem? Please describe.**
Both NLM Form Builder (https://lhcformbuilder.nlm.nih.gov/beta/) and Structor (https://formdesigner.helsenorgelab.no/) support adding help text via the questionnaire item control type `"help"` (see value-set http://hl7.org/fhir/R4/valueset-questionnaire-item-control.html). In addition, many LOINC forms loadable from the NLM Form Builder use help text. We should also prioritize supporting this control to have parity with these two popular tools.

Here's an example in use:

```json
"item": [
  {
    "text": "Sex of the patient, created for HL7 v3's \"Administrative gender\" concept, which represents the sex of a person used for adminstrative purposes (such as registering a patient in a hospital EHR). [https://www.hl7.org/fhir/v3/AdministrativeGender/index.html]",
    "type": "display",
    "linkId": "/72143-1-help",
    "extension": [
      {
        "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
        "valueCodeableConcept": {
          "coding": [
            {
              "system": "http://hl7.org/fhir/questionnaire-item-control",
              "code": "help",
              "display": "Help-Button"
            }
          ],
          "text": "Help-Button"
        }
      }
    ]
  }
]
```

**Describe the solution you'd like**
Support in SDC Library for the `"help"` item control type.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1528

**Description**

- Support item control extension for help code.
- If questionnaire item has help code then show help button
- Show and hide card view for help text on help button click
- Update all existing unit tests with material 3 theme to fix layout inflation exception.
- Update all existing android instrurmentation tests with material 3 to fix layout inflation exception.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**

Screenshot is attached just for demo purpose.

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@shelaghm 
Below material3 style is applied for Help (title and subtitle text in card view), as shared figma style was as per material2.
-  Text typography `TextAppearance.Material3.TitleSmall` and text color `colorOnSurfaceVariant` is applied for Help title text.
-  Text typography `TextAppearance.Material3.BodyMedium` and text color `colorOnSurfaceVariant` is applied for help subtitle (description) text.

**Comment by shelaghm** [comment]
@santosh-pingle those textstyles and colors look good. Thanks for checking with me. I have now updated the figma with the M3 style.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt [hunk 1]
```diff
@@ -141,6 +142,28 @@ internal val Questionnaire.QuestionnaireItemComponent.displayItemControl: Displa
     return DisplayItemControlType.values().firstOrNull { it.extensionCode == code }
   }
 
+/** Returns true if any one of the nested display item has [DisplayItemControlType.HELP] control. */
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt:145]:
```suggestion
/** Whether any one of the nested display item has [DisplayItemControlType.HELP] control. */
```

File: datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt [hunk 2]
```diff
@@ -141,6 +142,28 @@ internal val Questionnaire.QuestionnaireItemComponent.displayItemControl: Displa
...
+internal val Questionnaire.QuestionnaireItemComponent.hasHelpButton: Boolean
+  get() {
+    return item.any { it.isHelpCode }
+  }
+
+/**
+ * Returns [true] if item type is display and [displayItemControl] is
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt:152]:
```suggestion
 * Whether item type is display and [displayItemControl] is
```

File: datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt [hunk 3]
```diff
@@ -141,6 +142,28 @@ internal val Questionnaire.QuestionnaireItemComponent.displayItemControl: Displa
...
+ * [DisplayItemControlType.FLYOVER].
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt:153]:
```suggestion
 * [DisplayItemControlType.HELP].
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 4]
```diff
@@ -764,6 +884,131 @@ class MoreQuestionnaireItemComponentsTest {
...
+          text = "parent question text"
+          type = Questionnaire.QuestionnaireItemType.BOOLEAN
+        }
+      )
+
+    assertThat(questionItemList.first().localizedHelpSpanned).isNull()
+  }
+
+  @Test
+  fun `localizedHelpSpanned should returns help`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:902]:
```suggestion
  fun `localizedHelpSpanned should return help`() {
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 5]
```diff
@@ -764,6 +884,131 @@ class MoreQuestionnaireItemComponentsTest {
       .isEqualTo("flyover text")
   }
 
+  @Test
+  fun `localizedHelpSpanned should returns null if no help code`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:888]:
```suggestion
  fun `localizedHelpSpanned should return null if no help code`() {
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 6]
```diff
@@ -764,6 +884,131 @@ class MoreQuestionnaireItemComponentsTest {
...
+              }
+            )
+        }
+      )
+
+    assertThat(questionItemList.first().localizedHelpSpanned.toString()).isEqualTo("help text")
+  }
+
+  @Test
+  fun `localizedHelpSpanned does not match locale should returns help`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:933]:
```suggestion
  fun `localizedHelpSpanned not matching locale should return help`() {
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 7]
```diff
@@ -764,6 +884,131 @@ class MoreQuestionnaireItemComponentsTest {
...
+            )
+        }
+      )
+    Locale.setDefault(Locale.US)
+
+    assertThat(questionItemList.first().localizedHelpSpanned.toString()).isEqualTo("help text")
+  }
+
+  @Test
+  fun `localizedHelpSpanned returns localized text for matching locale`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:973]:
```suggestion
  fun `localizedHelpSpanned matching locale should return localized text`() {
```

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt [hunk 8]
```diff
@@ -69,3 +75,31 @@ internal fun getViewGroupVisibility(vararg view: TextView): Int {
...
+    if (questionnaireItem.hasHelpButton) {
+      VISIBLE
+    } else {
+      GONE
+    }
+  val helpCardView = view.findViewById<MaterialCardView>(R.id.helpCardView)
+  var isHelpCardViewVisible = true
+  helpButton.setOnClickListener {
+    helpCardView.visibility =
+      if (isHelpCardViewVisible) {
```
[the remark this answers is not in the document]
Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemHeaderView.kt]: done.

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**